src/App.tsx

```diff
@@ -1,14 +1,58 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setIsLoading(true);
+      try {
+        const loadedTodos = await getTodos();
+
+        setTodos(loadedTodos);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadTodos();
+  }, []);
+
+  const filteredTodos = todos
+    .filter(todo => {
+      if (selectFilter === 'active') {
+        return !todo.completed;
+      }
+
+      if (selectFilter === 'completed') {
+        return todo.completed;
+      }
+
+      return true;
+    })
+    .filter(todo => {
+      const isQuery = query.toLowerCase().trim();
```

Comment:
```suggestion
      const preparedQuery = query.toLowerCase().trim();
```
`is` and `has` prefixes only for boolean values